processor/src/debug.rs

```diff
@@ -114,6 +120,44 @@ impl VmStateIterator {
             (None, false)
         }
     }
+
+    //TODO: refactor such that we refactor common logic between `prev()` and `next` into a shared function.
+    pub fn prev(&mut self) -> Option<VmState> {
```

Comment:
Maybe we should make an issue for this so someone else can revisit it later?

Reply:
Yeah I think this is the most pragmatic approach.  Leave it as it is now and create an issue.  It allows us to focus on other more pressing matters.  I didn't want to spend too much time on this.  Wdyt @bobbinth ?

Reply:
I think creating an issue for factoring out common logic is fine. But I would like to have `DoubleEndedIterator` implemented in this PR if possible.

Reply:
yep, I'll get it done.